code/go/0chain.net/blobbercore/handler/object_operation_handler.go

```diff
@@ -447,6 +447,11 @@ func (fsh *StorageHandler) DownloadFile(
 		}
 
 		buyerEncryptionPublicKey := shareInfo.ClientEncryptionPublicKey
+
+		if common.Timestamp(shareInfo.AvailableAt.Unix()) > common.Now() {
```

Comment:
@Rizary 
I can see an issue here. Bcz unix timestamp is until milliseconds, it can cause error when customer trying to download let's say at 01.01.2022 12.00 - it will not let him download if milliseconds are upper.
Consider to check it up to minutes instead of seconds/milliseconds